File: examples/java/src/main/java/SVGQRCode.java
import io.github.g0dkar.qrcode.QRCode;
import io.github.g0dkar.qrcode.internals.QRCodeSquare;
import io.github.g0dkar.qrcode.render.Colors;
import qrcode.QRCode;
import qrcode.internals.QRCodeSquare;

import java.io.FileOutputStream;
import java.io.IOException;

public class SVGQRCode {

    public void createQRCode(String content) throws IOException {
        FileOutputStream fileOut = new FileOutputStream("java-svg.svg");

        QRCode qrCode = new QRCode(content);
        qrCode.setQrCodeGraphicsFactory(new SVGQRCodeGraphics.Factory());
        QRCodeSquare[][] computedQRCode = qrCode.encode();

        int computedSize = qrCode.computeImageSize(QRCode.DEFAULT_CELL_SIZE, QRCode.DEFAULT_MARGIN, computedQRCode);

        SVGQRCodeGraphics svgGraphics = new SVGQRCodeGraphics(computedSize, computedSize);

        qrCode.renderShadedIntoGraphics(QRCode.DEFAULT_CELL_SIZE, QRCode.DEFAULT_MARGIN, computedQRCode, svgGraphics, (qrCodeSquare, qrCodeGraphics) -> {
            if (qrCodeSquare.getDark()) {
                svgGraphics.fillRect(qrCodeSquare.absoluteX(), qrCodeSquare.absoluteY(), qrCodeGraphics.getWidth(), qrCodeGraphics.getHeight(), Colors.BLACK);
            }

            return null;
        });

        svgGraphics.writeImage(fileOut);
    }

    public static void main(String[] args) throws Exception {
        new SVGQRCode()
                .createQRCode("Hello, SVG!");
    }
}

File: examples/kotlin/src/main/kotlin/SVGQRCode.kt
import qrcode.color.Colors
import qrcode.raw.QRCodeBuilder
import java.io.FileOutputStream

class SVGQRCode {
    fun createQRCode(content: String) {
        FileOutputStream("kotlin-svg.svg").use {
            val qrCode = QRCodeBuilder(content)
            val computedSize = qrCode.computeImageSize()
            val graphics = SVGQRCodeGraphics(computedSize, computedSize)

            qrCode.renderShaded(qrCodeGraphics = graphics) { qrCodeSquare, cellGraphics ->
                if (qrCodeSquare.dark) {
                    graphics.fillRect(
                        qrCodeSquare.absoluteX(),
                        qrCodeSquare.absoluteY(),
                        cellGraphics.width,
                        cellGraphics.height,
                        Colors.BLACK
                    )
                }
            }

            graphics.writeImage(it)
        }
    }
}

fun main() {
    SVGQRCode()
        .createQRCode("Hello, SVG!")
}
